clientconn.go

```diff
@@ -1329,7 +1329,7 @@ func (ac *addrConn) createTransport(addr resolver.Address, copts transport.Conne
 	select {
 	case <-time.After(time.Until(connectDeadline)):
 		// We didn't get the preface in time.
-		newTr.Close()
+		newTr.Close(fmt.Errorf("failed to receive server preface within timeout"))
```

Comment:
I need to think about this more.. I don't see why this is here.  We already fail in the transport itself if the preface isn't received before the deadline.

https://github.com/grpc/grpc-go/blob/1d1bbb55b381f39fbe93edbb1d0fd96a6b1ecaef/internal/transport/http2_client.go#L1316

And also irrelevant here: I'm not really sure why we don't block in `NewClientTransport` until the preface is received, but I seem to remember there was some discussion about this a couple years ago when we last touched this code, so it may be intentional.

Reply:
ACK - I've left this alone for now, PLMK if want changes here